src/components/BlogPostCard.vue

```diff
@@ -0,0 +1,173 @@
+    <div class="post-content-container">
+      <div class="user-header-container">
+        <p class="user-name-text">{{ user.firstName }} {{ user.lastName }}</p>
+        <p class="user-email-text">&centerdot; {{ user.email }}</p>
+      </div>
+      <p class="post-content-text">{{ props.post.body }}</p>
+      <p class="tags-text" v-if="props.post?.tags?.length">
+        {{ allTags.join(' &centerdot; ') }}
+      </p>
+      <div class="actions-container">
+        <div
+          class="default-action-btn"
+          @click="toggleReaction"
+          :class="{ 'has-reacted': hasReacted }"
+        >
+          <FontAwesomeIcon :icon="[hasReacted ? 'fas' : 'far', 'heart']" />
+          <a onclick="">Reacts {{ post.reactions ? `(${post.reactions})` : '' }}</a>
+        </div>
+        <div
+          v-if="props.post?.userId === props.currentUserId"
+          class="default-action-btn"
+          @click="goToPostEdit"
+        >
+          <FontAwesomeIcon :icon="['far', 'pen-to-square']" />
+          <a onclick="">Edit</a>
+        </div>
+        <div v-if="props.post?.userId === props.currentUserId" class="delete-btn">
+          <FontAwesomeIcon :icon="['far', 'trash-can']" />
+          <a v-on:click="deletePost">Delete</a>
+        </div>
+      </div>
+    </div>
+    <div class="date-container">{{ postDateStr }}</div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue'
+import type { PropType } from 'vue'
+import type { BlogPost, BlogUser } from '@/store/types'
+import { useStore } from 'vuex'
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
+import { useRouter } from 'vue-router'
+
+const props = defineProps({
+  post: {
+    type: Object as PropType<BlogPost>,
+    required: true
+  },
+  currentUserId: {
+    type: String,
+    required: true
+  }
+})
+const store = useStore()
+const router = useRouter()
+
+const postDateStr = props.post?.createDate
+  ? new Date(props.post?.createDate).toLocaleDateString()
+  : ''
+const user = computed<BlogUser>(() => store.getters.usersById[props.post?.userId])
+const hasReacted = computed(
+  () => store.getters.reactionsByUser[props.currentUserId]?.[props.post?.postId]
+)
+const allTags = [
+  user.value.addressCity,
+  user.value.companyName,
+  ...(props.post?.tags ?? [])
+].filter((tag) => tag?.length)
+
+const toggleReaction = () => {
+  store.dispatch('changePostReact', { userId: props.currentUserId, postId: props.post?.postId })
+}
+const deletePost = () => {
+  if (props.post?.userId === props.currentUserId) {
+    store.dispatch('deleteBlogPost', props.post)
+  }
+}
+
+const goToPostEdit = () => {
+  if (props.post?.postId) {
+    router.push({
+      name: 'blog-post-edit',
+      params: { id: props.post?.postId }
+    })
+  }
+}
+</script>
+
+<style scoped lang="less">
+.card-container {
+  position: relative;
+  border-radius: 3px;
+  border: rgb(245, 246, 247) 2px solid;
```

Comment:
It would be good to define some of these colors in less variables

border: @lightBrown 2px solid;